docs/tutorials/build_a_node_in_java.md

```diff
@@ -92,26 +92,29 @@ BroadcasterInterface tx_broadcaster =
 
 **References:** [Rust docs](https://docs.rs/lightning/*/lightning/chain/chaininterface/trait.BroadcasterInterface.html), [Java bindings](https://github.com/lightningdevkit/ldk-garbagecollected/blob/main/src/main/java/org/ldk/structs/BroadcasterInterface.java)
 
-### 4. Optional: Initialize the `NetGraphMsgHandler`
+### 4. Optional: Initialize the `NetworkGraph`
 **You must follow this step if:** you need LDK to provide routes for sending payments (i.e. you are *not* providing your own routes)
 
 **What it's used for:** generating routes to send payments over
 
-**Example:** initializing `NetGraphMsgHandler` without providing an `Access`
+**Example:** initializing `NetworkGraph` on signet
 
 ```java
-final byte[] genesis_block_hash = // <insert the genesis block hash>
-final NetGraphMsgHandler router = NetGraphMsgHandler.of(
-    NetworkGraph.of(genesis_block_hash), Option_AccessZ.none(), logger);
+Network network = Network.LDKNetwork_Signet;
+
+BestBlock genesisBlock = BestBlock.from_genesis(network);
+final byte[] genesis_block_hash = genesisBlock.block_hash();
+
+final NetworkGraph networkGraph = NetworkGraph.of(genesis_block_hash);
 ```
 
-**Implementation notes:** this struct is not required if you are providing your own routes.
+**Implementation notes:** this struct is not required if you are providing your own routes. It will be used internally in `ChannelManagerConstructor` to build a `NetGraphMsgHandler`. Other networking options are: `LDKNetwork_Bitcoin`, `LDKNetwork_Regtest` and `LDKNetwork_Testnet`.
 
-**Dependencies:** `Logger`
+**Dependencies:** *none*
 
-**Optional dependency:** `Access`, a source of chain information. Recommended to be able to verify channels before adding them to the internal network graph.
+**Optional dependency:** `BestBlock`, the best known block as identified by its hash and height. Recommended to retrieve the genesis block from the target network.
```

Comment:
This shouldn't be an optional dependency, right? Not sure we need to state this, however, as the genesis block hash is what is needed.

This is a side effect of some recent refactoring. Prior to it, we only needed to pass `Network` constant. But afterwards, the block hash of the genesis block is needed. This unfortunately is not exported in the bindings, thus the need to use `BestBlock.from_genesis`.

Also, not really a recommendation as you *must* use the genesis block hash.